src/vocabulary/domains/LeitnerSystems.ts

```diff
@@ -41,15 +41,18 @@ export default class LeitnerSystems extends BaseEntityWithoutMandatoryId {
     }
 
     static calculateNextBoxAppearanceDate(box: LeitnerBoxType): Date {
-        const boxAppearanceDateMap: Record<LeitnerBoxType, Date> = {
-            [LeitnerBoxType.BOX_1]: new Date(),
-            [LeitnerBoxType.BOX_2]: makeItNewer(new Date(), MomentUnit.DAYS, LeitnerBoxAppearanceDifference.BOX_2),
-            [LeitnerBoxType.BOX_3]: makeItNewer(new Date(), MomentUnit.DAYS, LeitnerBoxAppearanceDifference.BOX_3),
-            [LeitnerBoxType.BOX_4]: makeItNewer(new Date(), MomentUnit.DAYS, LeitnerBoxAppearanceDifference.BOX_4),
-            [LeitnerBoxType.BOX_5]: new Date(),
+        const boxAppearanceDateMap: Record<LeitnerBoxType, DateFn> = {
```

Comment:
The problem with the previous solution is it is used to calculate the date for each type of box. The current solution calculates for the received box only!